src/Php/XdebugHandler.php

```diff
@@ -0,0 +1,185 @@
+            if ($this->prepareRestart()) {
+                $this->restart($this->getCommand());
+            }
+        }
+
+        if (self::RESTART_HANDLE === $args[0]) {
+            putenv(self::ENV_DISABLE_XDEBUG);
+
+            if (false !== $this->envScanDir) {
+                // $args[1] contains the original value
+                if (isset($args[1])) {
+                    putenv(ConfigBuilder::ENV_PHP_INI_SCAN_DIR . '=' . $args[1]);
+                } else {
+                    putenv(ConfigBuilder::ENV_PHP_INI_SCAN_DIR);
+                }
+            }
+        }
+    }
+
+    private function needsRestart(string $allow): bool
+    {
+        if (PHP_SAPI !== 'cli' || !\defined('PHP_BINARY')) {
+            return false;
+        }
+
+        return $this->isLoaded && empty($allow);
+    }
+
+    private function prepareRestart(): bool
+    {
+        if ($this->tmpIniPath = $this->configBuilder->build()) {
+            return $this->setEnvironment();
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns true if the restart environment variables were set
+     *
+     * @return bool
+     */
+    private function setEnvironment(): bool
+    {
+        return putenv(self::ENV_DISABLE_XDEBUG . '=' . self::RESTART_HANDLE);
+    }
+
+    /**
+     * Returns the restart command line
+     *
+     * @return string
+     */
+    private function getCommand(): string
+    {
+        $params = array_merge(
+            [PHP_BINARY, '-c', $this->tmpIniPath],
+            $this->getScriptArgs($_SERVER['argv'])
+        );
+
+        return implode(' ', array_map([$this, 'escape'], $params));
+    }
+
+    /**
+     * Returns the restart script arguments, adding --ansi if required
+     *
+     * If we are a terminal with color support we must ensure that the --ansi
+     * option is set, because the restarted output is piped.
+     *
+     * @param array $args The argv array
+     *
+     * @return array
+     */
+    private function getScriptArgs(array $args): array
+    {
+        if (\in_array(['--no-ansi', '--ansi'], $args, true)) {
+            return $args;
+        }
+
+        $offset = \count($args) > 1 ? 2 : 1;
+        array_splice($args, $offset, 0, '--ansi');
+
+        return $args;
+    }
+
+    /**
+     * Escapes a string to be used as a shell argument.
+     *
+     * From https://github.com/johnstevenson/winbox-args
+     * MIT Licensed (c) John Stevenson <john-stevenson@blueyonder.co.uk>
+     *
+     * @param string $arg  The argument to be escaped
+     * @param bool   $meta Additionally escape cmd.exe meta characters
+     *
+     * @return string The escaped argument
+     */
+    private function escape(string $arg, bool $meta = true): string
+    {
+        if (!\defined('PHP_WINDOWS_VERSION_BUILD')) {
+            return escapeshellarg($arg);
+        }
+
+        $quote = strpbrk($arg, " \t") !== false || $arg === '';
+        $arg = preg_replace('/(\\\\*)"/', '$1$1\\"', $arg, -1, $dquotes);
```

Comment:
`$quotesCount` instead of `$dquotes`? or does the `d` as a peculiar kind of meaning?